### PR TITLE
Evo 1654 add modal keyboard events handling

### LINKED ISSUE #1654: Add modal keyboard events handling
Problems:
1) The prev arrow is past the edge of the box and
2) the keyboard arrows don't work

### PR DESCRIPTION
**Pull request by KevinTriplett**
fixes #1654 

Problems:
- "prev arrow" icon of image carousel was off screen / clipped
- image carousel did not respond to left arrow and right arrow keyboard action
- modal did not respond to escape and return keyboard action
- clicking outside the modal did not dismiss the modal, an expected action for most users

Three commits:
- fix position of "prev arrow" on image carousel
- enable modal keyboard and mouse events
  - previously, only the background DOM received events
  - escape now cancels the modal
  - return now submits the modal (NOTE: if no submit action, just closes modal)
  - clicking anywhere outside the modal cancels it
- enable image carousel keyboard events
  - previously, only the background DOM received events
  - after above modal commit, slick-slide in image carousel still did not receive keyboard events
  - NOTE: only way to trigger slick-slide without a ref is to trigger mouse events on the arrow buttons

Best I could do quickly -- may have been a cleaner way to do it but eh

### CODE REVIEW

File: src/components/ModalDialog/ModalDialog.js [hunk 1]
```diff
@@ -61,13 +61,31 @@ class ModalDialog extends Component {
     useNotificationFormat: false
   }
 
+  modalRef = createRef()
+
+  handleKeydown = event => {
+    if (event.code === 'Escape') this.submit()
```
Comment (tibetsprague) [inline src/components/ModalDialog/ModalDialog.js:67]: uh shouldn't this be cancel?

File: src/components/ModalDialog/ModalDialog.js [hunk 2]
```diff
@@ -61,13 +61,31 @@ class ModalDialog extends Component {
...
+    if (event.code === 'Enter') this.cancel()
```
Comment (tibetsprague) [inline src/components/ModalDialog/ModalDialog.js:68]: and this submit?

**Review by tibetsprague — CHANGES_REQUESTED** [review]

File: src/components/ModalDialog/ModalDialog.js (hunk 2 above)
Comment (KevinTriplett) [reply src/components/ModalDialog/ModalDialog.js]: Ooh thanks ... yes, I recall that I originally was using event.keyCode and then switches to event.code ... and didn't switch the calls. Commit pushed

**Review by tibetsprague — APPROVED** [review]

**Comment by KevinTriplett** [comment]
I'm looking at tests I can add to improve coverage and catch future regressions

**Comment by KevinTriplett** [comment]
@tibetsprague if you want to look over the recent commits, I simplified the keyboard event handling tremendously and had to update some other files to get eslint to pass.

Which I assume means maybe you and Thomas don't have eslint automatically check work before pushing?

**Pull request merged**